client/navigation/NavigationStore.js

```diff
@@ -0,0 +1,31 @@
+
+/**
+ * Module dependencies.
+ */
+
+var createStore = require('dispatchr/utils/createStore');
+
+module.exports = createStore({
+    initialize: function () {
+    	this.state = {
+            currentPage: null
+        };
+    },
+    storeName: 'NavigationStore',
+    handlers: {
+        'NAVIGATE': function (route) {
+            var store = this.dispatcher.getStore('NavigationStore');
+            store.state.currentPage = route.name;
+            store.emit('NAVIGATE');
+        }
+    },
+    getCurrentPage: function () {
+        return this.state.currentPage;
+    },
+    dehydrate: function () {
+    	return this.state;
+    },
+    rehydrate: function (state) {
+    	this.state = state;
+    }
+});
```

Comment:
Why don't I clean that up in the same commit as fixing the indentation, because it's an issue that's scattered across the repo.